nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/PutMarkLogic.java

```diff
@@ -397,23 +421,48 @@ public final void onTrigger(final ProcessContext context, final ProcessSessionFa
      * batch size will be flushed immediately and not have to wait for more FlowFiles to arrive to fill out the batch.
      */
     public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
-        FlowFile flowFile = session.get();
+        final FlowFile flowFile = session.get();
         if (flowFile == null) {
             getLogger().debug("Flushing the WriteBatcher asynchronously in case a number of documents less than batchSize are waiting to be written");
             flushWriteBatcherAsync(this.writeBatcher);
             getLogger().debug("Calling yield() on the ProcessContext");
             context.yield();
-        } else {
-            try {
-                String duplicateHandler = context.getProperty(DUPLICATE_URI_HANDLING).getValue();
-                WriteEvent writeEvent = buildWriteEvent(context, session, flowFile);
+            return;
+        }
+
+        if (this.writeBatcher.isStopped()) {
```

Comment:
This is the key addition - check to see if the batcher has stopped, as it's useless if so, which then makes the processor useless. The processor will try to connect to MarkLogic first - there's no point in creating a new batcher if it cannot connect - and if so, it will create a new batcher, which then effectively brings the processor back to life.